grails-app/services/org/bbop/apollo/PermissionService.groovy

```diff
@@ -653,4 +653,21 @@ class PermissionService {
         handleToken(params, payloadJson)
         return payloadJson
     }
+
+    JSONObject handleInput(HttpServletRequest request, Map params) {
```

Comment:
There should be a bunch of calls to:

                JSONObject returnObject = permissionService.handleInput(request, params)  

which should mirror this.